File: apps/web/src/lib/types/vehicle.ts
export interface Vehicle {
  id: string;
  dealership_id: string;
  vin: string;
  stock_number: string | null;
  year: number | null;
  make: string | null;
  model: string | null;
  trim: string | null;
  mileage: number | null;
  status: string | null;
  source_price: string | null;
  displayed_price: string | null;
  website_verified_price: string | null;
  price_type: string | null;
  source_type: string | null;
  source_url: string | null;
  vehicle_url: string | null;
  image_url: string | null;
  price_verified: boolean;
  sync_status: string | null;
  created_at: string;
  updated_at: string;
}

export interface VehicleWithAssignment extends Vehicle {
  assigned_esl: string | null;
  assignment_id: string | null;
}

File: apps/api/app/models/vehicle.py
import uuid
from datetime import datetime
from decimal import Decimal

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, Numeric, Text, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class Vehicle(Base):
    __tablename__ = "vehicles"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True)
    dealership_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("dealerships.id", ondelete="CASCADE"), nullable=False
    )
    vin: Mapped[str] = mapped_column(Text, nullable=False)
    stock_number: Mapped[str | None] = mapped_column(Text)
    year: Mapped[int | None] = mapped_column(Integer)
    make: Mapped[str | None] = mapped_column(Text)
    model: Mapped[str | None] = mapped_column(Text)
    trim: Mapped[str | None] = mapped_column(Text)
    mileage: Mapped[int | None] = mapped_column(Integer)
    status: Mapped[str | None] = mapped_column(Text)
    source_price: Mapped[Decimal | None] = mapped_column(Numeric(12, 2))
    displayed_price: Mapped[Decimal | None] = mapped_column(Numeric(12, 2))
    website_verified_price: Mapped[Decimal | None] = mapped_column(Numeric(12, 2))
    price_type: Mapped[str | None] = mapped_column(Text)
    source_type: Mapped[str | None] = mapped_column(Text)
    source_url: Mapped[str | None] = mapped_column(Text)
    vehicle_url: Mapped[str | None] = mapped_column(Text)
    image_url: Mapped[str | None] = mapped_column(Text)
    last_source_update_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    last_website_verified_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    price_verified: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    sync_status: Mapped[str | None] = mapped_column(Text)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
